azkaban-common/src/main/java/azkaban/executor/ExecutorManager.java

```diff
@@ -1421,58 +1422,56 @@ private void finalizeFlows(ExecutableFlow flow) {
     // the reference.
 
     updaterStage = "finalizing flow " + execId + " alerting and emailing";
-    ExecutionOptions options = flow.getExecutionOptions();
-    // But we can definitely email them.
-    Alerter mailAlerter = alerters.get("email");
-    if (flow.getStatus() == Status.FAILED || flow.getStatus() == Status.KILLED) {
-      if (options.getFailureEmails() != null
-          && !options.getFailureEmails().isEmpty()) {
-        try {
-          mailAlerter.alertOnError(flow);
-        } catch (Exception e) {
-          logger.error(e);
-        }
-      }
-      if (options.getFlowParameters().containsKey("alert.type")) {
-        String alertType = options.getFlowParameters().get("alert.type");
-        Alerter alerter = alerters.get(alertType);
-        if (alerter != null) {
+    if(alertUser) {
+      ExecutionOptions options = flow.getExecutionOptions();
+      // But we can definitely email them.
+      Alerter mailAlerter = alerters.get("email");
+      if (flow.getStatus() == Status.FAILED || flow.getStatus() == Status.KILLED) {
+        if (options.getFailureEmails() != null && !options.getFailureEmails().isEmpty()) {
           try {
-            alerter.alertOnError(flow);
+            mailAlerter.alertOnError(flow);
           } catch (Exception e) {
-            // TODO Auto-generated catch block
-            e.printStackTrace();
-            logger.error("Failed to alert by " + alertType);
+            logger.error(e);
```

Comment:
this is just exception in sending emails. logging the exception should be ok. If we raise the same exception,  the appropriate way to handle it by downstream handler is log it anyway.